### ISSUE TITLE
Unhandled ErrorException in YouTube Provider

### ISSUE THREAD

**Issue by MegaGoblin** [SocialiteProviders/Providers#394]
```
protected function getUserByToken($token)
{
        $response = $this->getHttpClient()->get(
            'https://www.googleapis.com/youtube/v3/channels?part=snippet&mine=true', [
            'headers' => [
                'Authorization' => 'Bearer '.$token,
            ],
        ]);

        return json_decode($response->getBody()->getContents(), true)['items'][0];
 }
```

Situation: User have google account, and don't have youtube-account (not created yet)

Than:
ErrorException is occurs  in `return` on getting ['items'][0]
`items` is empty and i have ErrorExceprion with message `Undefined offset: 0`

**Comment by atymic** [comment]
Thanks for the bug report. We'll look into this shortly!

**Comment by robin7331** [comment]
Any progress on this ?

**Comment by fauxvo** [comment]
I did some experimenting and it also seems to be related to users who have not create a YouTube channel.

This wouldn't throw an error anymore after they created a channel.

It seems a bit clunky to find out the user info only via the channel information.  Maybe something more specific to Google would be better?  It would also be great to get actually get the user's email address since that's not really discussed either even though I thought I added the email scopes.

But yes, any progress?

**Comment by atymic** [comment]
@fauxvo 

Could you post the raw error that you get?
I think the issue is definatly due to the user not having a channel, but what's the expected behaviour in this case? I can make it return and empty user object, but is this ideal?

**Comment by fauxvo** [comment]
```
[2020-05-27 19:47:56] production.ERROR: Undefined index: items {"exception":"[object] (ErrorException(code: 0): Undefined index: items at /var/www/youplayparty.com/releases/44/vendor/socialiteproviders/youtube/Provider.php:56)
[stacktrace]
#0 /var/www/youplayparty.com/releases/44/vendor/socialiteproviders/youtube/Provider.php(56): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /var/www/youplayparty.com/releases/44/vendor/socialiteproviders/manager/src/OAuth2/AbstractProvider.php(44): SocialiteProviders\\YouTube\\Provider->getUserByToken()
#2 /var/www/youplayparty.com/releases/44/app/Http/Controllers/Auth/RegisterController.php(128): SocialiteProviders\\Manager\\OAuth2\\AbstractProvider->user()
#3 [internal function]: App\\Http\\Controllers\\Auth\\RegisterController->handleProviderCallback()
```

Basically, the `items` index is undefined and causing a 500 error.

I don't really know what the best solution is to be honest.  I'm not sure if there's another endpoint (probably a Google one rather than a Youtube one) that would return the user information - ie. access tokens, username, email, etc.  In my case, I'm using the YouTube login to be able to get the correct permissions to create a Playlist on the user's behalf.  I get that maybe there's a limitation on Google's side that won't let a user perform certain actions unless they go through the Channel creation process.  Maybe there's a way that the YouTube provider could throw an exception that the channel isn't setup vs a straight out 500?

**Comment by shehi** [comment]
I don't even think this Provider should exist in first place. Google has one central OAuth-2 implementation and that Provider is already covered. Why bother with this superficial package, if in a nutshell it's just Google OAuth-2?!

**Comment by atymic** [comment]
Good point @shehi feel free to PR a deprecation.

**Comment by atymic** [comment]
Should be fixed now in #572
